packages/protocol/PaintsCache.ts

```diff
@@ -41,17 +41,16 @@ export async function findFirstMeaningfulPaint() {
       const paint = paints[index];
 
       try {
-        const { value } = await StreamingScreenShotCache.readAsync(
-          replayClient,
-          paint.time,
-          paint.point
-        );
-        if (value && value.hash) {
-          const { width, height } = await getDimensions(value.hash, value.mimeType);
+        const screenShot = paint.paintHash
+          ? await screenshotCache.readAsync(replayClient, paint.point, paint.paintHash)
+          : undefined;
+
+        if (screenShot && screenShot.hash) {
+          const { width, height } = await getDimensions(screenShot.hash, screenShot.mimeType);
```

Comment:
This needs to use `screenShot.data`.
```suggestion
          const { width, height } = await getDimensions(screenShot.data, screenShot.mimeType);
```

Reply:
Ooof, yes.